plugins/analytics-module-ga4/src/apis/implementations/AnalyticsApi/GoogleAnalytics.ts

```diff
@@ -0,0 +1,269 @@
+      virtualSearchPageView,
+      testMode,
+    } = options;
+
+    this.cdmConfig = cdmConfig;
+    this.virtualSearchPageView = virtualSearchPageView;
+
+    // Initialize Google Analytics.
+    ReactGA.initialize(measurementId, {
+      testMode,
+    });
+
+    // If identity is required, defer event capture until identity is known.
+    this.capture = new DeferredCapture({ defer: identity === 'required' });
+
+    // Allow custom userId transformation.
+    this.customUserIdTransform =
+      typeof userIdTransform === 'function' ? userIdTransform : undefined;
+
+    // Capture user only when explicitly enabled and provided.
+    if (identity !== 'disabled' && identityApi) {
+      this.setUserFrom(identityApi);
+    }
+  }
+
+  /**
+   * Instantiate a fully configured GA Analytics API implementation.
+   */
+  static fromConfig(
+    config: Config,
+    options: {
+      identityApi?: IdentityApi;
+      userIdTransform?:
+        | 'sha-256'
+        | ((userEntityRef: string) => Promise<string>);
+    } = {},
+  ) {
+    // Get all necessary configuration.
+    const measurementId = config.getString('app.analytics.ga.measurementId');
+    const identity =
+      config.getOptionalString('app.analytics.ga.identity') || 'disabled';
+    const virtualSearchPageView = parseVirtualSearchPageViewConfig(
+      config.getOptionalConfig('app.analytics.ga.virtualSearchPageView'),
+    );
+    const testMode =
+      config.getOptionalBoolean('app.analytics.ga.testMode') ?? false;
+    const cdmConfig =
+      config
+        .getOptionalConfigArray('app.analytics.ga.customDimensionsMetrics')
+        ?.map(c => {
+          return {
+            type: c.getString('type') as CustomDimensionOrMetricConfig['type'],
+            index: c.getNumber('index'),
+            source: c.getString(
+              'source',
+            ) as CustomDimensionOrMetricConfig['source'],
+            key: c.getString('key'),
+          };
+        }) ?? [];
+
+    if (identity === 'required' && !options.identityApi) {
+      throw new Error(
+        'Invalid config: identity API must be provided to deps when ga.identity is required',
+      );
+    }
+
+    // Return an implementation instance.
+    return new GoogleAnalytics({
+      ...options,
+      identity,
+      measurementId,
+      virtualSearchPageView,
+      cdmConfig,
+      testMode,
+    });
+  }
+
+  /**
+   * Primary event capture implementation. Handles core navigate event as a
+   * pageview and the rest as custom events. All custom dimensions/metrics are
+   * applied as they should be (set on pageview, merged object on events).
+   */
+  captureEvent(event: AnalyticsEvent) {
+    const { context, action, subject, value, attributes } = event;
+    const customMetadata = this.getCustomDimensionMetrics(context, attributes);
+
+    if (action === 'navigate' && context.extension === 'App') {
+      this.capture.pageview(subject, customMetadata);
+      return;
+    }
```

Comment:
Similar to how we have custom handling for the `navigate` event, it might be worth adding some custom handling for the `click` event.  Based on [the docs here](https://support.google.com/analytics/answer/9234069?hl=en#click), it looks like we should be capturing the `subject` (link text) on the `link_text` default dimension, and we could similarly be populating `link_url`, `link_domain` and `outbound` properties based on the `to` attribute.